pyproject.toml

```diff
@@ -1,7 +1,64 @@
 [build-system]
-requires = ["setuptools>=61", "wheel", "setuptools_scm>=7.0.3"]
+requires = ["setuptools>=64", "wheel", "setuptools_scm>=8"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "pyradiosky"
+authors = [
+    {name = "Bryna Hazelton", email = "brynah@phys.washington.edu"},
+    {name = "Matthew Kolopanis", email = "mjkolopa@asu.edu"},
+    {name = "Jonathan Pober", email = "jonathan_pober@brown.edu"},
+    {name = "Adam Lanman", email = "alanman@mit.edu"},
+]
+maintainers = [
+    {name = "Bryna Hazelton", email = "brynah@phys.washington.edu"},
+    {name = "Matthew Kolopanis", email = "mjkolopa@asu.edu"}
+]
+description = "Python objects and interfaces for representing diffuse, extended and compact astrophysical radio sources"
+readme = "README.md"
+dynamic = ["version"]
+dependencies = [
+    "astropy>=6.0",
+    "h5py>=3.4",
+    "numpy>=1.23",
+    "pyuvdata>=2.4.3",
+    "scipy>=1.7.3",
+    "setuptools>=61",
+    "setuptools_scm>=7.0.3",
+]
+requires-python = ">=3.10"
+keywords = ["radio astronomy"]
+classifiers = [
+    "License :: OSI Approved :: BSD License",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering :: Astronomy",
+]
+
+[project.optional-dependencies]
+healpix = ["astropy-healpix>=1.0.2"]
+astroquery = ["astroquery>=0.4.4"]
+lunarsky = ["lunarsky>=0.2.2"]
+all = ["pyradiosky[healpix,astroquery,lunarsky]"]
+test = ["coverage", "pre-commit", "pytest", "pytest-cov"]
+doc = ["matplotlib", "pypandoc", "sphinx"]
+dev = ["pyradiosky[all,test,doc]"]
+
+[project.urls]
+Repository = "https://github.com/RadioAstronomySoftwareGroup/pyradiosky"
+Documentation = "https://pyradiosky.readthedocs.io/"
+
+# Note that this approach is "discouraged" in favor of [project.scripts], but
+# that requires entry-point functions, which would require some refactoring.
```

Comment:
I do highly recommend it though!

Reply:
I second this! but otherwise also feel like this PR is good. And am a ruff advocate as well.

Reply:
I think we should do this, I was just going to do it in a different PR.

Reply:
Same with ruff.

Reply:
I made issues for both of these so we won't lose track of them.